crates/linker/src/lib.rs

```diff
@@ -473,25 +460,60 @@ pub fn preprocess(
         println!("Targeting: {}", target);
     }
 
-    if let target_lexicon::BinaryFormat::Coff = target.binary_format {
-        crate::pe::preprocess_windows(
-            exec_filename,
-            Path::new(metadata_filename),
-            Path::new(out_filename),
-            shared_lib,
-            verbose,
-            time,
-        )
-        .unwrap_or_else(|e| internal_error!("{}", e));
+    match target.binary_format {
+        target_lexicon::BinaryFormat::Elf | target_lexicon::BinaryFormat::Macho => {
+            preprocess_elf_and_macho(
+                target,
+                host_exe_path,
+                metadata_path,
+                preprocessed_path,
+                shared_lib,
+                verbose,
+                time,
+            );
+        }
 
-        return;
+        target_lexicon::BinaryFormat::Coff => {
+            crate::pe::preprocess_windows(
+                host_exe_path,
+                metadata_path,
+                preprocessed_path,
+                verbose,
+                time,
+            )
+            .unwrap_or_else(|e| internal_error!("{}", e));
+        }
+
+        target_lexicon::BinaryFormat::Wasm => {
```

Comment:
This shouldn't actually come up? They would use the linker for wasm that brian wrote? Or is that not the case if we use the llvm backend?